nodes.py

```diff
@@ -43,1848 +46,474 @@ def interrupt_processing(value=True):
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            return (None,) 
+
+        config = {
+            "api_key": api_key,
+            "base_url": base_url,
+            "deployment_name": deployment_name,
+            "model_name": model_name,
+            "openai_api_version": openai_api_version,
+        }
+        return (config,)
 
-class VAEEncodeForInpaint:
+class GroqConfigModelNode:
     @classmethod
     def INPUT_TYPES(s):
-        return {"required": { "pixels": ("IMAGE", ), "vae": ("VAE", ), "mask": ("MASK", ), "grow_mask_by": ("INT", {"default": 6, "min": 0, "max": 64, "step": 1}),}}
-    RETURN_TYPES = ("LATENT",)
-    FUNCTION = "encode"
-
-    CATEGORY = "latent/inpaint"
-
-    def encode(self, vae, pixels, mask, grow_mask_by=6):
-        x = (pixels.shape[1] // vae.downscale_ratio) * vae.downscale_ratio
-        y = (pixels.shape[2] // vae.downscale_ratio) * vae.downscale_ratio
-        mask = torch.nn.functional.interpolate(mask.reshape((-1, 1, mask.shape[-2], mask.shape[-1])), size=(pixels.shape[1], pixels.shape[2]), mode="bilinear")
-
-        pixels = pixels.clone()
-        if pixels.shape[1] != x or pixels.shape[2] != y:
-            x_offset = (pixels.shape[1] % vae.downscale_ratio) // 2
-            y_offset = (pixels.shape[2] % vae.downscale_ratio) // 2
-            pixels = pixels[:,x_offset:x + x_offset, y_offset:y + y_offset,:]
-            mask = mask[:,:,x_offset:x + x_offset, y_offset:y + y_offset]
-
-        #grow mask by a few pixels to keep things seamless in latent space
-        if grow_mask_by == 0:
-            mask_erosion = mask
-        else:
-            kernel_tensor = torch.ones((1, 1, grow_mask_by, grow_mask_by))
-            padding = math.ceil((grow_mask_by - 1) / 2)
-
-            mask_erosion = torch.clamp(torch.nn.functional.conv2d(mask.round(), kernel_tensor, padding=padding), 0, 1)
-
-        m = (1.0 - mask.round()).squeeze(1)
-        for i in range(3):
-            pixels[:,:,:,i] -= 0.5
-            pixels[:,:,:,i] *= m
-            pixels[:,:,:,i] += 0.5
-        t = vae.encode(pixels)
-
-        return ({"samples":t, "noise_mask": (mask_erosion[:,:,:x,:y].round())}, )
-
+        return {
+            "required": {
+                "groq_api_key": ("STRING", {"multiline": False, "default": ""}),
+                "model_name": ("STRING", {"multiline": False, "default": ""}),
+                "temperature": ("FLOAT", {"default": 0.7, "min": 0.0, "max": 1.0, "step": 0.1}),
+            },
+        }
+    
+    RETURN_TYPES = ("LLM",)
+    FUNCTION = "configure"
+    CATEGORY = "LLM Config"
+
+    def configure(self, groq_api_key, model_name, temperature):
+        config = {
+            "groq_api_key": groq_api_key,
+            "model_name": model_name,
+            "temperature": temperature,
+        }
+        return (config,)
 
-class InpaintModelConditioning:
+class GeminiConfigModelNode:
```

Comment:
These classes not need we will be using the type of llm provider so these llm configurations classes not required.